### PR TITLE
Rebase to 057

### PR DESCRIPTION
**Pull request by pvalena**
Actual WIP rebase to 057 + additional commits.

The layered commits can be seen here:

**updated once more** [[PR: 057 <- master]](https://github.com/redhat-plumbers/dracut-rhel9/pull/40)

_ _ _ _

### PR readiness check-list

 - [x] Reviewed layered commits (PR above)
 - [x] Reviewed whole diff+commits
 - [x] github CI passed
 - [x] Tested build in a VM ([log](https://gist.github.com/f5b881a74c6c1b1a990329b6d3cfc87f))
 - [x] Sent reference to some BZ bugs (for testing)
 - [x] osci functional tests (internal) passed
 - [x] Run test suite in a VM (**some failures:** [log](https://gist.github.com/3e2896bfbc0d7e2d3b3208465ca85e50))
 - [x] Peer-Reviewed layered commits
 - [x] Push the update to Rawhide

_ _ _ _ 

**Note: this PR is only for review purposes. I will force push and current main will stay as `old-main` branch (https://github.com/redhat-plumbers/dracut-rhel9/commit/32524aa0f72f9e63198927c22a9dcf37b497d538).** The result is identical to pushing upstream tag `057` and merging this PR (cleanly).

### CODE REVIEW

**Comment by pvalena** [comment]
> @pvalena maybe rebase to main branch will help.

This is intended to be force pushed :) it has some commits from this *fork* overlayed on upstream *tag* 057 (release).

**Comment by pvalena** [comment]
I found out what the issue was .... I used main instead of master ... (on the https://github.com/pvalena/dracut-rhel9/pull/1 PR)... now they're the same

**Comment by pvalena** [comment]
CI won't help us in case of merge conflicts :/.
Ideas?

**Comment by pvalena** [comment]
Please do not change yet, as I do intend to go through the scratch-builds and all changes from 057.

**Comment by mrc0mmand** [comment]
Yeah, CIs won't help you there, since they're using the `refs/pull/XXX/merge` ref, so the PR branch must be conflict-free.

**Comment by pvalena** [comment]
> > CI won't help us in case of merge conflicts :/. Ideas?
> 
> I think you need to resolve them first, GA doesn't like conflicts. /cc @mrc0mmand

Like I said, this is going to be force-push, as we're changing the  *base* to upstream tag with newer release. There have to be conflicts. Also, I'm squashing all CI changes and applying them in a single commit. Please see updated https://github.com/pvalena/dracut-rhel9/pull/2.

**Comment by pvalena** [comment]
Basically, resetting the repo to upstream commit, and applying our downstream changes on top of it :).

**Comment by mrc0mmand** [comment]
Well, you can try pushing a `new-main` branch to this repository with the rebased main, and change the base branch for this PR to the `new-main` branch. That should make CIs happy.

**Comment by pvalena** [comment]
> Well, you can try pushing a `new-main` branch to this repository with the rebased main, and change the base branch for this PR to the `new-main` branch. That should make CIs happy.

This is what I did in my fork, which is a PR that I've referred to as "layered commits". But that probably didn't trigger the same automation. Let me try to reopen here as well then :).

Thanks.

**Comment by pvalena** [comment]
> > Like I said, this is going to be force-push, as we're changing the _base_ to upstream tag with newer release. There have to be conflicts. Also, I'm squashing all CI changes and applying them in a single commit. Please see updated [pvalena#2](https://github.com/pvalena/dracut-rhel9/pull/2).
> 
> Sorry, I didn't get it at first, but still, I thing you are missing at least [32524aa](https://github.com/redhat-plumbers/dracut-rhel9/commit/32524aa0f72f9e63198927c22a9dcf37b497d538)

Thanks for checking! Let me find where the mistake happened...

**Comment by pvalena** [comment]
> > > Like I said, this is going to be force-push, as we're changing the _base_ to upstream tag with newer release. There have to be conflicts. Also, I'm squashing all CI changes and applying them in a single commit. Please see updated [pvalena#2](https://github.com/pvalena/dracut-rhel9/pull/2).
> > 
> > 
> > Sorry, I didn't get it at first, but still, I thing you are missing at least [32524aa](https://github.com/redhat-plumbers/dracut-rhel9/commit/32524aa0f72f9e63198927c22a9dcf37b497d538)
> 
> Thanks for checking! Let me find where the mistake happened...

I'm not! :) I just squashed all the CI commits (there was a lot of them; didn't make much sense for review).... I've diffed it (and will once more; all commits) .... and I hope i'm not missing any changes.

While doing a rebase onto 057, there were conflict, off course, which I had to resolve manually, which means the commits wouldn't match 1:1 anyway - so diff of relevant files might be actually simpler than going through all the (amended) commits.

File: .github/workflows/differential-shellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+
+    steps: 
+      - name: Repository checkout
+        uses: actions/checkout@v2
+
+      - name: Differential ShellCheck
+        uses: redhat-plumbers-in-action/differential-shellcheck@v1
```
Comment (pvalena) [inline .github/workflows/differential-shellcheck.yml:17]: Ah, I figured it out- 
Seems I forgot to fetch latest commits .... 

```
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Total 5 (delta 4), reused 4 (delta 4), pack-reused 0
Unpacking objects: 100% (5/5), 582 bytes | 291.00 KiB/s, done.
From github.com:redhat-plumbers/dracut-rhel9
   1cd36263..32524aa0  main       -> plumbers-9/main

```

Comment (pvalena) [reply .github/workflows/differential-shellcheck.yml:17]: Which is actually just small change which was merged in the meantime:

```diff
diff --git a/.github/workflows/differential-shellcheck.yml b/.github/workflows/differential-shellcheck.yml
index 095c3fe6..55273b3d 100644
--- a/.github/workflows/differential-shellcheck.yml
+++ b/.github/workflows/differential-shellcheck.yml
@@ -1,7 +1,5 @@
 name: Differential ShellCheck
 on:
-  push:
-    branches: [main]
   pull_request:
     branches: [main]

@@ -11,7 +9,11 @@ jobs:

     steps:
       - name: Repository checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0

       - name: Differential ShellCheck
-        uses: redhat-plumbers-in-action/differential-shellcheck@v1
+        uses: redhat-plumbers-in-action/differential-shellcheck@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}

```

Comment (pvalena) [reply .github/workflows/differential-shellcheck.yml:17]: On that topic, do we need `b/.github/labeler.yml` ?

Also this change I'm not sure what to do about:
https://gist.github.com/afedcdea1b691f90e7bca56f0aa676d3
(most probably comes from upstream)

Comment (pvalena) [reply .github/workflows/differential-shellcheck.yml]: @jamacku WDYT?

Comment (pvalena) [reply .github/workflows/differential-shellcheck.yml]: What do you think? (sorry; abbrev)

But I guess current state (as long as it works), is ok.... What I was asking is about

 - `.github/labeler.yml` - as I've noticed there were some downstream changes previously (I don't remember from the top of my head; probably removed)
 - https://gist.github.com/afedcdea1b691f90e7bca56f0aa676d3
   - IOW: `.github/workflows/integration.yml` and
   - `.github/workflows/container.yml`

As I have done some manual changes, due to conflicts with upstream (or what did make sense for me). 

I hope those files are OK :]

File: modules.d/95resume/module-setup.sh [hunk 2]
```diff
@@ -5,15 +5,15 @@ check() {
...
+            block_is_netdevice "$(get_maj_min "$_dev")" && return 0
         done
         return 1
     }
 
     # Only support resume if hibernation is currently on
     # and no swap is mounted on a net device
     [[ $hostonly ]] || [[ $mount_needs ]] && {
-        swap_on_netdevice || [[ "$(cat /sys/power/resume)" == "0:0" ]] && return 255
+        swap_on_netdevice || [[ -f /sys/power/resume && "$(cat /sys/power/resume)" == "0:0" ]] && return 255
```
Comment (pvalena) [inline modules.d/95resume/module-setup.sh:16]: lnykryn ^

File: .github/workflows/differential-shellcheck.yml (hunk 1 above)
Comment (mrc0mmand) [reply .github/workflows/differential-shellcheck.yml]: The Fedora stuff was taken from the upstream directly, I added only the C8S stuff, so bumping Fedora to F34/F35 would definitely make sense.

Comment (pvalena) [reply .github/workflows/differential-shellcheck.yml]: I've synced (some of) the changes from upstream (there's no longer 34/33, but just 'latest'), along with the newly created files.

Comment (pvalena) [reply .github/workflows/differential-shellcheck.yml]: Thanks for checking!

**Review by jamacku — COMMENTED** [review]
I think we don't want `.github/stale.yml`.

**Comment by pvalena** [comment]
> I think we don't want `.github/stale.yml`.

I quite like it :) it sends comments to PRs which are not active and you simply need to set it to Bug /  Doc / Regression to disble it :). I think that'll work for me...

Do you want me to drop it anyway?

**Comment by pvalena** [comment]
FTR: [squashing CI commits] backed up as https://github.com/pvalena/dracut-rhel9/tree/rhel-9-rebase-ci-commits

**Comment by jamacku** [comment]
> > I think we don't want `.github/stale.yml`.
> 
> I quite like it :) it sends comments to PRs which are not active and you simply need to set it to Bug / Doc / Regression to disble it :). I think that'll work for me...
> 
> Do you want me to drop it anyway?

If you like it, I don't mind :+1:

**Pull request merged**